File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/report/CharacterReportMessage.scala





















// Generated on 05/03/2013 10:34:59
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait CharacterReportMessage extends MessageDeserializer {
  val messageId = 6079

  protected trait Message extends BaseMessage with Serializable {
    val messageId = CharacterReportMessage.messageId

    val reportedId: UInt
    val reason: Byte

    def serialize(buf: Buffer) {
      buf.writeUInt(reportedId)
      buf.writeByte(reason)
    }

  }

  protected def create(reportedId: UInt, reason: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val reportedId = buf.readUInt()
    val reason = buf.readByte()

    create(reportedId, reason)
  }

}

object CharacterReportMessage extends CharacterReportMessage {
  case class MessageImpl(reportedId: UInt, reason: Byte) extends Message
  type Target = MessageImpl

  override def create(reportedId: UInt, reason: Byte) = MessageImpl(reportedId, reason)

  def apply(reportedId: UInt, reason: Byte) = create(reportedId, reason)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/CharacterReportMessage.java


// Generated on 05/08/2013 19:37:58
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class CharacterReportMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6079;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public long reportedId;
    public byte reason;
    
    public CharacterReportMessage() { }
    
    public CharacterReportMessage(long reportedId, byte reason) {
        this.reportedId = reportedId;
        this.reason = reason;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeUInt(reportedId);
        buf.writeByte(reason);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        reportedId = buf.readUInt();
        if (reportedId < 0 || reportedId > 4294967295L)
            throw new RuntimeException("Forbidden value on reportedId = " + reportedId + ", it doesn't respect the following condition : reportedId < 0 || reportedId > 4294967295L");
        reason = buf.readByte();
        if (reason < 0)
            throw new RuntimeException("Forbidden value on reason = " + reason + ", it doesn't respect the following condition : reason < 0");
    }
    
}
